### PR TITLE
LongestRepeatingSequence

### PR DESCRIPTION
**Pull request by Himsahu**
The longest sequence which has been repeated twice

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**

* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Java files are placed inside an existing directory.
* [ ] All filenames are in all uppercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Java naming conventions.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/LongestRepeatingSequence [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        for(int i = 0; i < n; i++){  
+            for(int j = i+1; j < n; j++){  
+                
+                String x = lcp(str.substring(i,n),str.substring(j,n));  
+                if(x.length() > lrs.length()) lrs=x;  
+            }  
+        }  
+        System.out.println("Longest repeating sequence: "+lrs);  
+    }  
+}
```
Comment (mertcandav) [inline strings/LongestRepeatingSequence:28]: Hi. Whitespace patterns in the code are unbalanced. Please keep the number of spaces between operators and tokens.
```suggestion
public class LongestRepeatingSequence {
    //Checks for the largest common prefix
    public static String lcp(String s, String t) {
        int n = Math.min(s.length(), t.length());
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) != t.charAt(i)) {
                return s.substring(0, i);
            }  
        }  
        return s.substring(0,n);  
    }  
  
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.next();
        String lrs="";
        int n = str.length();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                String x = lcp(str.substring(i, n), str.substring(j, n));
                if(x.length() > lrs.length()) {
                    lrs = x;
                }
            }
        }
        System.out.println("Longest repeating sequence: " + lrs);
    }
}
```

**Review by mertcandav — CHANGES_REQUESTED** [review]

**Comment by mertcandav** [comment]
Please add comments for the algorithm to be understood by all.

File: strings/LongestRepeatingSequence (hunk 1 above)
Comment (Himsahu) [reply strings/LongestRepeatingSequence]: Please accept all my pull request, so that I can participate in Hacktoberfest

**Review by mertcandav — APPROVED** [review]
LGTM 👍🏼

**Pull request closed without merging**